internal/repositories/validate.go

```diff
@@ -0,0 +1,55 @@
+package repositories
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/DIMO-Network/telemetry-api/internal/graph/model"
+	"github.com/DIMO-Network/telemetry-api/internal/service/ch"
+)
+
+var (
+	twoWeeks       = 14 * 24 * time.Hour
+	errInvalidArgs = errors.New("invalid arguments")
+)
+
+func validateAggSigArgs(args *model.AggregatedSignalArgs) error {
+	if args.FromTS.IsZero() {
+		return fmt.Errorf("%w from timestamp is zero", errInvalidArgs)
+	}
+	if args.ToTS.IsZero() {
+		return fmt.Errorf("%w to timestamp is zero", errInvalidArgs)
+	}
+
+	// check if time range is greater than 2 weeks
+	if args.ToTS.Sub(args.FromTS) > twoWeeks {
+		return fmt.Errorf("%w time range is greater than 2 weeks", errInvalidArgs)
+	}
+
+	if time.Duration(args.Interval) < 1 {
```

Comment:
I thought `Interval` was in milliseconds. `time.Duration` is milliseconds, so this whole thing reads a bit odd.